File: Algorithms/Strings/Strong Password.py
import os

#
# Complete the 'minimumNumber' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER n
#  2. STRING password
#

def minimumNumber(n, password):
    # Return the minimum number of characters to make the password strong
    count = 0
    criteria = 0

    if not any(char.islower() for char in password):
        criteria += 1
    if not any(char.isupper() for char in password):
        criteria += 1
    if not any(char.isdigit() for char in password):
        criteria += 1
    if not any(char in "!@#$%^&*()-+" for char in password):
        criteria += 1

    count = max(criteria, 6 - n)

    return count

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input().strip())
    password = input()
    answer = minimumNumber(n, password)

    fptr.write(str(answer) + '\n')
    fptr.close()

File: Algorithms/Strings/Strong Password.php
<?php

/*
 * Complete the 'minimumNumber' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts following parameters:
 *  1. INTEGER n
 *  2. STRING password
 */

function minimumNumber($n, $password) {
    # Return the minimum number of characters to make the password strong
    $digitPresent = False;
    $uppercasePresent = False;
    $lowercasePresent = False;
    $specialPresent = False;

    $specialChars = "!@#$%^&*()-+";

    for ($i = 0; $i < $n; $i++) {
        $char = $password[$i];

        if (is_numeric($char)) {
            $digitPresent = True;
        }
        elseif (ctype_upper($char)) {
            $uppercasePresent = True;
        }
        elseif (ctype_lower($char)) {
            $lowercasePresent = True;
        }
        elseif (strpos($specialChars, $char) !== False) {
            $specialPresent = True;
        }
    }

    $requiredCount = 0;

    if (!$digitPresent) {
        $requiredCount++;
    }

    if (!$uppercasePresent) {
        $requiredCount++;
    }

    if (!$lowercasePresent) {
        $requiredCount++;
    }

    if (!$specialPresent) {
        $requiredCount++;
    }

    $additionalChars = max(0, 6 - $n);

    return max($requiredCount, $additionalChars);
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$n = intval(trim(fgets(STDIN)));
$password = rtrim(fgets(STDIN), "\r\n");
$answer = minimumNumber($n, $password);

fwrite($fptr, $answer . "\n");
fclose($fptr);
